core/lib/zksync_core/src/api_server/web3/pubsub.rs

```diff
@@ -3,23 +3,41 @@
 use std::{collections::HashMap, sync::Arc};
 
 use anyhow::Context as _;
-use jsonrpc_core::error::{Error, ErrorCode};
-use jsonrpc_pubsub::{typed, SubscriptionId};
 use tokio::{
     sync::{mpsc, watch, RwLock},
     task::JoinHandle,
     time::{interval, Duration},
 };
 use zksync_dal::ConnectionPool;
 use zksync_types::{MiniblockNumber, H128, H256};
-use zksync_web3_decl::types::{BlockHeader, Log, PubSubFilter, PubSubResult};
+use zksync_web3_decl::{
+    jsonrpsee::{
+        core::{server::SubscriptionMessage, SubscriptionResult},
+        server::IdProvider,
+        types::{error::ErrorCode, ErrorObject, SubscriptionId},
+        PendingSubscriptionSink, SubscriptionSink,
+    },
+    namespaces::EthPubSubServer,
+    types::{BlockHeader, Log, PubSubFilter, PubSubResult},
+};
 
 use super::{
     metrics::{SubscriptionType, PUB_SUB_METRICS},
     namespaces::eth::EVENT_TOPIC_NUMBER_LIMIT,
 };
 
-pub(super) type SubscriptionMap<T> = Arc<RwLock<HashMap<SubscriptionId, T>>>;
+#[derive(Debug, Clone, Copy, Default)]
+#[non_exhaustive]
```

Comment:
Why do you need `#[non_exhaustive]` here? The struct itself can have reduced visibility (e.g., `pub(super)`), in which case `non_exhaustive` particularly doesn't make sense (it is only effective for public data types).

Reply:
removed